### ISSUE TITLE
Account Balances in Bank Reconciliation Tool

### ISSUE THREAD

**Issue by marination** [frappe/erpnext#36998]
> **Summary**: **What balance is `get_account_balance` really supposed to return ?**

Assume:
- I have a bank account called 'HDFC - R' which is impacted via final entries (PE, JV, etc.)
- This account will be untouched by Invoices since they make intermediate GL Entries
- This account is linked to a "Bank Account" record in my system.

In https://github.com/frappe/erpnext/blob/develop/erpnext/accounts/doctype/bank_reconciliation_tool/bank_reconciliation_tool.py#L59-L83 :
- `balance_as_per_system` is nullified by `total_debit` and `total_credit`.
-  `get_balance_on()` essentially gets `sum(debit - credit)` until a certain date
-  So assuming : `total debit = 12700` and `total_credit = 240`, `get_balance_on() = 12460`
- But 12460 - total_debit + total_credit = 0
- So get_account_balance() always gives me a 0. But Trial Balance or event the Account Tree shows me 12460 as the balance. 

So here if all my GL Entries resulted from Payment + Journal Entries, the result would be 0. 

_I would expect `get_account_balance` to give me the balance of 'HDFC - R' (bank account that has final impact after making payment) which would be **12460**_

### Module

accounts

### Version

ERPNext - v14 (latest)

**Comment by nissekalle0** [comment]
The same we have discovered.

**Comment by ZhangLianK** [comment]
I also face the same issue. get_balance_on function has get the balance from general ledger, so I think there is no need to get payment entry

**Comment by this-gavagai** [comment]
I am also struggling with very erroneous values in the Bank Reconciliation Tool. I am confused as to why all the complexity is needed. Shouldn't the bank balance match the general ledger balance for any given date? The problem with selecting just payments and journal entries is that other transaction types can also affect bank balance, such as Sales Invoice, Purchase Invoice, Loan Repayment, etc.

**Comment by NobleDruid** [comment]
I agree. The reconciliation tool should just pull from the ledger balances for dates selected on the reconciliation tool. Also, it is possible the discrepancy with reconciliation balances may be related to this other issue:
https://github.com/frappe/erpnext/issues/33932
